rectools/models/popular.py

```diff
@@ -41,17 +43,83 @@ class Popularity(Enum):
     SUM_WEIGHT = "sum_weight"
 
 
+def _serialize_timedelta(td: tp.Optional[tp.Union[None, dict, timedelta]]) -> tp.Optional[timedelta]:
+    if isinstance(td, dict):
+        return timedelta(
+            days=td.get("days", 0),
+            seconds=td.get("seconds", 0),
+            microseconds=td.get("microseconds", 0),
+            milliseconds=td.get("milliseconds", 0),
+            minutes=td.get("minutes", 0),
+            hours=td.get("hours", 0),
+            weeks=td.get("weeks", 0),
+        )
+    return td
+
+
+def _deserialize_timedelta(td: tp.Optional[timedelta]) -> tp.Optional[dict]:
+    if td is None:
+        return td
+    return {"days": td.days, "seconds": td.seconds, "microseconds": td.microseconds}
+
+
+TimeDelta = tpe.Annotated[
+    tp.Union[None, timedelta, dict],
+    PlainValidator(func=_serialize_timedelta),
+    PlainSerializer(func=_deserialize_timedelta),
+]
+
+
 class PopularModelConfig(ModelConfig):
     """Config for `PopularModel`."""
 
     popularity: Popularity = Popularity.N_USERS
-    period: tp.Optional[timedelta] = None
-    begin_from: tp.Optional[datetime] = None
+    period: TimeDelta = None
+    begin_from: tp.Optional[tp.Union[datetime, str]] = None
     add_cold: bool = False
     inverse: bool = False
 
 
-class PopularModel(FixedColdRecoModelMixin, ModelBase):
+class PopularModelMixin:
+    """Mixin for models based on popularity."""
+
+    def _validate_popular_model_attributes(
+        self,
+        popularity: tp.Literal["n_users", "n_interactions", "mean_weight", "sum_weight"],
+        period: TimeDelta,
+        begin_from: tp.Optional[tp.Union[datetime, str]],
+    ) -> None:
+        try:
+            self.popularity = Popularity(popularity)
```

Comment:
`Explicit is better than implicit` is one of python principles. It's a very bad approach to change inner attribute in method named `validate...`. Better return it from here.

Also let's split this method into `_validate_popularity` and `_validate_<i can't invent the name right now, help me please>`

Reply:
Done.
Only validation process was left here.